src/dbnode/storage/series/buffer.go

```diff
@@ -500,18 +475,205 @@ func (b *dbBuffer) FetchBlocksMetadata(
 			Size:     resultSize,
 			LastRead: resultLastRead,
 		})
-	})
+	}
 
 	return res
 }
 
-type dbBufferBucket struct {
-	opts              Options
+func (b *dbBuffer) newBucketsAt(
+	t time.Time,
+) *dbBufferBuckets {
+	buckets := b.bucketsPool.Get()
+	buckets.resetTo(t, b.opts, b.bucketPool)
+	b.bucketsMap[xtime.ToUnixNano(t)] = buckets
```

Comment:
All the places where this is called will check this doesn't exist already presumably?